### ISSUE TITLE
[Bug]: [JSON] Redundant data in returned entity when specifying JSON field as output fields in query

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24164]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230516-ccd68501
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0.dev30
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Redundant data in returned entity when specifying JSON field as output fields in query
each returned result res[0], res[1]... has all the searched json data.

```
res= collection.query(expr="json_field[\"number\"] <= 10", output_fields=["json_field"])
res[0]
{'json_field': [b'{"number":0,"string":"milvus","bool":true,"list":[]}', b'{"number":2,"string":"milvus","bool":true,"list":[0,1]}', b'{"number":4,"string":"milvus","bool":true,"list":[0,1,2,3]}', b'{"number":6,"string":"milvus","bool":true,"list":[0,1,2,3,4,5]}', b'{"number":8,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7]}', b'{"number":10,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7,8,9]}', b'{"number":1,"string":"milvus","bool":true,"list":[0]}', b'{"number":3,"string":"milvus","bool":true,"list":[0,1,2]}', b'{"number":5,"string":"milvus","bool":true,"list":[0,1,2,3,4]}', b'{"number":7,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6]}', b'{"number":9,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7,8]}'], 'int64': 0}

res[1]
{'json_field': [b'{"number":0,"string":"milvus","bool":true,"list":[]}', b'{"number":2,"string":"milvus","bool":true,"list":[0,1]}', b'{"number":4,"string":"milvus","bool":true,"list":[0,1,2,3]}', b'{"number":6,"string":"milvus","bool":true,"list":[0,1,2,3,4,5]}', b'{"number":8,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7]}', b'{"number":10,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7,8,9]}', b'{"number":1,"string":"milvus","bool":true,"list":[0]}', b'{"number":3,"string":"milvus","bool":true,"list":[0,1,2]}', b'{"number":5,"string":"milvus","bool":true,"list":[0,1,2,3,4]}', b'{"number":7,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6]}', b'{"number":9,"string":"milvus","bool":true,"list":[0,1,2,3,4,5,6,7,8]}'], 'int64': 1}
```

### Expected Behavior

NO redundant data in returned entity when specifying JSON field as output fields in query
res[0]
{'json_field': [b'{"number":0,"string":"milvus","bool":true,"list":[]}'], 'int64': 0}

res[1]
{'json_field': [b'{"number":1,"string":"milvus","bool":true,"list":[0]}'], 'int64': 1}
### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

connections.connect()
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, float_vector])
collection = Collection("test_search_collection_binbin_tmp", schema=schema)
import random
nb=3000
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
import numpy as np
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)]} for i in range(0, nb)], vectors])
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
collection.flush()
collection.create_index("float_vector", index_param, index_name="index_name_1")
collection.load()
res= collection.query(expr="json_field[\"number\"] <= 10", output_fields=["json_field"])
res[0]
res[1]
```

**Comment by yah01** [comment]
/assign @binbinlv 
Fixed with latest PyMilvus milvus-io/pymilvus#1435

**Comment by binbinlv** [comment]
Verified and fixed:
pymilvus: 2.4.0.dev31
```
>>> res= collection.query(expr="json_field[\"number\"] <= 10", output_fields=["json_field"])
>>> res[0]
{'json_field': b'{"number":0,"string":"milvus","bool":true,"list":[]}', 'int64': 0}
>>>
>>>
>>> res[1]
{'json_field': b'{"number":1,"string":"milvus","bool":true,"list":[0]}', 'int64': 1}
>>> res[2]
{'json_field': b'{"number":2,"string":"milvus","bool":true,"list":[0,1]}', 'int64': 2}
>>> res[3]
{'json_field': b'{"number":3,"string":"milvus","bool":true,"list":[0,1,2]}', 'int64': 3}
```

**Comment by binbinlv** [comment]
This issue for 2.2 verification is blocked by issue https://github.com/milvus-io/milvus/issues/24144, will try after https://github.com/milvus-io/milvus/issues/24144 fixed to check whether this exists in 2.2.* branch too.

Reopen this for the tracking on 2.2.*.

**Comment by binbinlv** [comment]
Verified and fixed:
milvus: [2.2.0-20230520-1eb804fc](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230520-1eb804fc/images/sha256-f1fce0b4ba76324119d89cebe860886197da341061e6688f0ef9da34719ef485?context=explore)
pymilvus: 2.2.9.dev12

```
>>> res= collection.query(expr="json_field[\"number\"] <= 10", output_fields=["json_field"])
>>> res[0]
{'json_field': b'{"number":0,"string":"milvus","bool":true,"list":[]}', 'int64': 0}
>>> res[1]
{'json_field': b'{"number":1,"string":"milvus","bool":true,"list":[0]}', 'int64': 1}
```
